### PR TITLE
Updated Responsive Header and Footer, Updated Cart Component, Fixed Styling Issues

### PR DESCRIPTION
**Pull request by special-PPT**
# Pull Request Template

## Description
Please provide a detailed description of your pull request. Include its purpose, what problem it solves, etc.

## Type of Change
Please check the options that are relevant to your PR:
- [x] Bug fix
- [ ] New feature
- [ ] Documentation update
- [x] Code refactor
- [ ] Other, please describe:

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes:
1. 
2. 
3. 

## Checklist
Before submitting your PR, please make sure you have completed the following steps:
- [x] I have tested my changes locally and they do not break existing functionality.
- [x] I have updated the documentation accordingly.
- [x] I follow the code style of this project.

## Additional Information
If applicable, add any other information that could help the reviewers.

### CODE REVIEW

File: src/front-end/src/styles/main/global.css [hunk 1]
```diff
@@ -5,9 +5,10 @@
     /*===== Colors =====*/
     --primary-color-dark: #111827;
     --primary-color-white: #ffffff;
+    --primary-color-blue: #5048E5;
+    --primary-color-gray: #373A3C;
+    --primary-color-light-gray: #979797;
     --header-color-yellow: #FCE944;
-    --button-color-blue: #5048E5;
```
Comment (yanruc123) [inline src/front-end/src/styles/main/global.css:9]: 这个我在用，不用删

File: src/front-end/src/styles/main/global.css [hunk 2]
```diff
@@ -5,9 +5,10 @@
...
-    --link-color-blue: #5048E5;
```
Comment (yanruc123) [inline src/front-end/src/styles/main/global.css:10]: 这个也是

**Review by yanruc123 — CHANGES_REQUESTED** [review]
global里的2个color不要删，我css里用到了

File: src/front-end/src/styles/main/global.css (hunk 2 above)
Comment (special-PPT) [reply src/front-end/src/styles/main/global.css:10]: 好的，新提交了一个commit，加了这两个colors

**Comment by malker97** [comment]
这个好像是你俩有交叉的文件，所以，我把我从这个PR的reviewer list中移除了

**Comment by special-PPT** [comment]
> 这个好像是你俩有交叉的文件，所以，我把我从这个PR的reviewer list中移除了

好的

**Review by yanruc123 — APPROVED** [review]

**Pull request merged**